src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -124,6 +125,8 @@ public class StorageService extends NotificationBroadcasterSupport implements IE
     private final List<ProgressListener> bootstrapListeners = new CopyOnWriteArrayList<>();
 
     private final Condition startBootstrapCondition = new SimpleCondition(DISABLE_WAIT_TO_BOOTSTRAP);
+    // we expect to use startRequestStreamsCondition to gate decommissions as well with `.reset()`
```

Comment:
Nit: `// TODO(dguo): use startRequestStreamsCondition to gate ...`
for easier search